ios/FluentUI/Pill Button Bar/PillButton.swift

```diff
@@ -9,84 +9,43 @@ import UIKit
 @objc(MSFPillButton)
-open class PillButton: UIButton, FluentUIWindowProvider {
-
-    /// Set `backgroundColor` to customize background color of the pill button
-    @objc open var customBackgroundColor: UIColor? {
-        didSet {
-            updateAppearance()
-        }
-    }
-
-    /// Set `selectedBackgroundColor` to customize background color of the pill button
-    @objc open var customSelectedBackgroundColor: UIColor? {
-        didSet {
-            updateAppearance()
-        }
-    }
-
-    /// Set `textColor` to customize background color of the pill button
-    @objc open var customTextColor: UIColor? {
-        didSet {
-            updateAppearance()
-        }
-    }
-
-    /// Set `selectedTextColor` to customize background color of the pill button
-    @objc open var customSelectedTextColor: UIColor? {
-        didSet {
-            updateAppearance()
-        }
-    }
-
-    /// Set `unreadDotColor` to customize color of the pill button unread dot
-    @objc open var customUnreadDotColor: UIColor? {
-        didSet {
-            updateAppearance()
-        }
-    }
+open class PillButton: UIButton, TokenizedUIControlInternal, ControlConfiguration {
 
     open override func didMoveToWindow() {
         super.didMoveToWindow()
 
-        pillButtonTokens.updateForCurrentTheme()
+        updatePillButtonTokens()
         updateAppearance()
     }
 
     @objc public init(pillBarItem: PillButtonBarItem, style: PillButtonStyle = .primary) {
         self.pillBarItem = pillBarItem
         self.style = style
-        self.pillButtonTokens = MSFPillButtonTokens(style: style)
         super.init(frame: .zero)
-        pillButtonTokens.windowProvider = self
         setupView()
 
-        pillButtonTokens.themeDidUpdate = { [weak self] in
-            guard let strongSelf = self else {
-                return
-            }
-            strongSelf.unreadDotLayer = strongSelf.initUnreadDotLayer()
-            strongSelf.updateAppearance()
-        }
-
         NotificationCenter.default.addObserver(self,
                                                selector: #selector(isUnreadValueDidChange),
                                                name: PillButtonBarItem.isUnreadValueDidChangeNotification,
                                                object: pillBarItem)
     }
 
-    var unreadDotColor: UIColor = Colors.gray100
-
-    @objc public static let cornerRadius: CGFloat = 16.0
-
-    @objc public let pillBarItem: PillButtonBarItem
-
-    @objc public let style: PillButtonStyle
-
     public required init?(coder aDecoder: NSCoder) {
         preconditionFailure("init(coder:) has not been implemented")
     }
 
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        updateUnreadDot()
+    }
+
+    public func overrideTokens(_ tokens: PillButtonTokens?) -> Self {
+        overrideTokens = tokens
+        return self
+    }
+
+    public typealias TokenType = PillButtonTokens
```

Comment:
I don't believe this line should be necessary, even though Xcode sometimes suggests it is.